### ISSUE TITLE
Training highway-v0 with DQN, the trained result is not as good as the one presented in the example video

### ISSUE THREAD

**Issue by limeng-1234** [eleurent/highway-env#225]
1. **example code**

import gym
import highway_env
from stable_baselines import DQN
model = DQN('MlpPolicy', "highway-fast-v0",
              policy_kwargs=dict(net_arch=[256, 256]),
              learning_rate=5e-4,
              buffer_size=15000,
              learning_starts=200,
              batch_size=32,
              gamma=0.8,
              train_freq=1,
              gradient_steps=1,
              target_update_interval=50,
              verbose=1,
              tensorboard_log="highway_dqn/")
model.learn(int(2e4))
model.save("highway_dqn/model")

Question：
1、highway-fast-v0 does not seem to exist
2、the result is not good as example
3、These parameters, such as gradient_steps, target_update_interval, do not exist in stable_baselines.DQN

**Comment by eleurent** [comment]
Hi @limeng-1234,

1. It was introduced in the last release, you need to upgrade `highway-env`
2. I just realised I used the wrong video in the docs, sorry. I'll update it. The actual DQN(MLP) performance are a little bit worse.
3. Yes sorry again, it should be stable_baselines3.DQN. I'll fix the script.

**Comment by limeng-1234** [comment]
sorry，i use windows operation system and i don't know how to use the latest highway_env

**Comment by eleurent** [comment]
> 
> 
> sorry，i use windows operation system and i don't know how to use the latest highway_env

Open a python prompt (or Anaconda if applicable), and type
```
pip install highway-env --upgrade
```
That will give you the last stable release.
You can also install the development version, see https://highway-env.readthedocs.io/en/latest/installation.html#development-version

**Comment by eleurent** [comment]
I fixed the docs for your questions 2. and 3., and 1. should work after you update.
Feel free to reopen if it still does not work.

**Comment by limeng-1234** [comment]
Thank you for the detailed reply, and there is still something wrong

Code: import gym
import highway_env
from stable_baselines3 import DQN


importgymimporthighway_envfromstable_baselines3importDQNmodel=DQN('MlpPolicy',"highway-fast-v0",policy_kwargs=dict(net_arch=[256,256]),learning_rate=5e-4,buffer_size=15000,learning_starts=200,batch_size=32,gamma=0.8,train_freq=1,gradient_steps=1,target_update_interval=50,verbose=1,tensorboard_log="highway_dqn/")model.learn(int(2e4))model.save("highway_dqn/model")# Load and test saved modelmodel=DQN.load("highway_dqn/model")whileTrue:done=Falseobs=env.reset()whilenotdone:action,_states=model.predict(obs,deterministic=True)obs,reward,done,info=env.step(action)env.render()
comment: The environment is not packaged as an 'env' in the last part. And it is will be better if you can provide the correct agent's results through DQN


| |
李蒙
|
|
***@***.***
|
签名由网易邮箱大师定制
On 9/9/2021 15:29，Edouard ***@***.***> wrote：

Closed #225.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by eleurent** [comment]
> The environment is not packaged as an 'env' in the last part

Sorry, I did not understand that, can you elaborate?

> And it is will be better if you can provide the correct agent's results through DQN

You're right, I will add my training curves.

**Comment by eleurent** [comment]
>    The environment is not packaged as an 'env' in the last part

Ah, I see. Sorry again about that.
Please refer to the the script that I use: https://github.com/eleurent/highway-env/blob/master/scripts/sb3_highway_dqn.py
I will just copy its content to the docs.

**Comment by limeng-1234** [comment]
Thank you very much. thank you thank you thank you


| |
limeng
|
|
***@***.***
|
签名由网易邮箱大师定制
On 9/9/2021 16:10，Edouard ***@***.***> wrote：

The environment is not packaged as an 'env' in the last part

Ah, I see. Sorry again about that.
Please refer to the the script that I use: https://github.com/eleurent/highway-env/blob/master/scripts/sb3_highway_dqn.py
I will just copy its content to the docs.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by eleurent** [comment]
> Thank you very much. thank you thank you thank you

Thank *you* for your feedback! I uploaded my training results to the docs: https://highway-env.readthedocs.io/en/latest/quickstart.html

**Comment by limeng-1234** [comment]
After training with your new code, the training results are better than before, although there are still collisions and abnormal behavior.

Sorry for another question, I would like to ask you if you have trained in the continuous action space.





| |
李蒙
|
|
***@***.***
|
签名由网易邮箱大师定制
On 9/9/2021 17:10，Edouard ***@***.***> wrote：

Thank you very much. thank you thank you thank you

Thank you for your feedback! I uploaded my training results to the docs: https://highway-env.readthedocs.io/en/latest/quickstart.html

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by eleurent** [comment]
> After training with your new code, the training results are better than before, although there are still collisions and abnormal behavior.

You should get the same performance/behaviour as I do, as that reported in https://highway-env.readthedocs.io/en/latest/quickstart.html#training-an-agent
You can check it by running tensorboard.

To obtain better performances, see the note below the results in the documentation.

> Sorry for another question, I would like to ask you if you have trained in the continuous action space.

Only in the `parking-v0` environment, I have not tried it for other environments such as highway.
But other people have, you should be able to find these attempts in the closed issues.
